### PR TITLE
Migrate to Log4j 2

### PR DESCRIPTION
**Pull request by philipp94831**

### CODE REVIEW

File: src/test/resources/log4j2.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration>
+  <Appenders>
+    <Console name="Console" target="SYSTEM_OUT">
+      <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
+    </Console>
+  </Appenders>
+  <Loggers>
+    <Logger name="com.bakdata" level="debug" additivity="false">
```
Comment (yannick-roeder) [inline src/test/resources/log4j2.xml:9]: This was info before. Is debug intended here?

File: src/test/resources/log4j2.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      <AppenderRef ref="Console"/>
+    </Logger>
+    <Root level="info">
```
Comment (yannick-roeder) [inline src/test/resources/log4j2.xml:12]: This was warning before. Is info intended here?

File: src/test/resources/log4j2.xml (hunk 1 above)
Comment (philipp94831) [reply src/test/resources/log4j2.xml:9]: There was no test logging config before. Only production but that was never used

Comment (yannick-roeder) [reply src/test/resources/log4j2.xml:9]: Ok, I thought it is the replacement for `log4j.logger.com.bakdata=INFO`

File: src/main/resources/log4j.properties [hunk 3]
```diff
@@ -1,5 +0,0 @@
-log4j.rootLogger=WARN, console
```
Comment (yannick-roeder) [inline src/main/resources/log4j.properties:1]: What happened to this logging config then if the other one is a test config?

Comment (philipp94831) [reply src/main/resources/log4j.properties:1]: As I said: This was never used anywhere because this is a library and not a docker artifact

**Review by yannick-roeder — APPROVED** [review]

**Pull request merged**